src/app/services/bungie-api.service.ts

```diff
@@ -339,120 +289,101 @@ export class BungieApiService {
+        // remove sunset items
+        let instanceData = profile.Response.itemComponents.instances.data || {};
+        let instance = instanceData[d.itemInstanceId || ""] || {};
+        return !!instance.energy;
+      })
+      .map((d) => {
+        let instanceData = profile.Response.itemComponents.instances.data || {};
+        let instance = instanceData[d.itemInstanceId || ""] || {};
+
+        if (!res[d.itemHash]) {
+          console.warn("Missing manifest item for item hash", d.itemHash);
+          return null;
+        }
+
+        let r = createArmorItem(
+          res[d.itemHash],
+          d.itemInstanceId || "",
+          InventoryArmorSource.Inventory
+        );
+        r.masterworked = !!instance.energy && instance.energy.energyCapacity == 10;
+        r.energyLevel = !!instance.energy ? instance.energy.energyCapacity : 0;
+        const sockets = profile.Response.itemComponents.sockets.data || {};
+        const socketsList =
+          sockets[d.itemInstanceId!]?.sockets.map((socket) => socket.plugHash) ?? [];
+        collectInvestmentStats(r, res[d.itemHash]?.investmentStats ?? [], socketsList, mods);
+
+        // Take a look if it really has the artifice perk
+        if (r.perk == ArmorPerkOrSlot.SlotArtifice) {
+          let statData = profile.Response.itemComponents.perks.data || {};
+          let perks = (statData[d.itemInstanceId || ""] || {})["perks"] || [];
+          const hasPerk = perks.filter((p) => p.perkHash == 229248542).length > 0;
+          if (!hasPerk) r.perk = ArmorPerkOrSlot.None;
+        }
+
+        if (!r.isExotic && this.config_assumeEveryLegendaryIsArtifice)
+          r.perk = ArmorPerkOrSlot.SlotArtifice;
 
-          if (!r.isExotic && this.config_assumeEveryLegendaryIsArtifice)
-            r.perk = ArmorPerkOrSlot.SlotArtifice;
+        return r as IInventoryArmor;
+      })
+      .filter(Boolean) as IInventoryArmor[];
+
+    // Now add the collection rolls for exotics
+    const collectionRollItems = Array.from(unlockedExoticArmorItemHashes)
+      .map((exoticItemHash) => {
+        const manifestArmorItem = res[exoticItemHash];
+        if (!manifestArmorItem) {
+          console.error("Couldn't find manifest item for exotic", exoticItemHash);
+          return null;
+        }
+
+        const collectionItem = createArmorItem(
+          manifestArmorItem,
+          `c${manifestArmorItem.hash}`,
+          InventoryArmorSource.Collections
+        );
+
+        collectInvestmentStats(
+          collectionItem,
+          manifestArmorItem.investmentStats,
+          manifestArmorItem.socketEntries.map((s) => s.singleInitialItemHash),
+          mods
+        );
+
+        return collectionItem;
+      })
+      .filter(Boolean) as IInventoryArmor[];
 
-          return r as IInventoryArmor;
-        }) || [];
+    r = r.concat(collectionRollItems);
 
     r = r.filter((k) => !k["statPlugHashes"] || k["statPlugHashes"][0] != null);
 
+    r = r.concat(vendorArmorItems);
```

Comment:
I just wonder, why is this below the filter? Do the vendor rolls not have statPlugHashes?

Reply:
Correct, since the stats for the vendor items are read directly from the `ItemStats` component, we don’t have the plug hashes available.